nltk/parse/dependencygraph.py

```diff
@@ -321,10 +326,15 @@ def extract_10_cells(cells):
                         'CoNLL(10) or Malt-Tab(4) format'.format(cell_number)
                     )
 
-            word, lemma, ctag, tag, feats, head, rel = cell_extractor(cells)
+            line_index, word, lemma, ctag, tag, feats, head, rel = cell_extractor(cells)
+            try:
+                index = int(line_index)
+            except TypeError: pass  # line_index is probably None, use default
+            except ValueError: pass  # line_index is not an integer, use default
 
+            if head == '_': continue
             head = int(head)
-            if zero_based:
+            if zero_based and line_index is None:
                 head += 1
 
             self.nodes[index].update(
```

Comment:
Yes, you're right. Changed it to replace index, which I think is easier to read anyway.